### PR TITLE
Minor changes for clipping

### LINKED ISSUE #50: Dealing with model collapse
**Is your feature request related to a problem? Please describe.**
Sometimes, because of the way a model gets initialized/trained, *model collapse* (i.e., loss and weights all become `nan`) can happen.

**Describe the solution you'd like**
Instead of having re-initialize the complete training sequence, we should explore ways to make GaNDLF automatically do this.

**Describe alternatives you've considered**
Open to other suggestions.

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes #50 and includes #85

## Proposed Changes

- function renamed to add `_` at the end to highlight in-place operation
- unit test added
- testing covers clip separately
- AMP has clipping as optional

## Checklist

- [x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [x] Non-breaking change (would not break existing functionality): if changes breaks current code, please provide as many details as possible
- [x] Function/class source code documentation added/updated
- [x] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [x] If applicable, version information [has been updated in GANDLF/version.py](../GANDLF/version.py)
- [x] If adding a submodule, add to list of expections for black styling in [pyproject.toml](../pyproject.toml) file 
- [x] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [x] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [x] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`

### CODE REVIEW

File: GANDLF/misc_utils/clip_gradients.py [hunk 1]
```diff
@@ -4,10 +4,10 @@
 """
 
 import torch
-from GANDLF.misc_utils.adaptive_gradient_clipping import adaptive_gradient_clip
+from GANDLF.misc_utils.adaptive_gradient_clip_ping import adaptive_gradient_clip_
```
Comment (Geeks-Sid) [inline GANDLF/misc_utils/clip_gradients.py]: Please correct the import. You probably used a find and replace.

**Review by Geeks-Sid — CHANGES_REQUESTED** [review]

File: GANDLF/misc_utils/clip_gradients.py (hunk 1 above)
Comment (Geeks-Sid) [reply GANDLF/misc_utils/clip_gradients.py]: Please pytest this once. :)

Comment (sarthakpati) [reply GANDLF/misc_utils/clip_gradients.py]: This is outdated... [This](https://github.com/sarthakpati/GaNDLF/blob/master/GANDLF/misc_utils/clip_gradients.py#L7) is the latest version... 😄

File: GANDLF/training_loop.py [hunk 2]
```diff
@@ -139,7 +139,10 @@ def train_network(model, train_dataloader, optimizer, params):
     # automatic mixed precision - https://pytorch.org/docs/stable/amp.html
     if params["model"]["amp"]:
         print("Using Automatic mixed precision", flush=True)
-        scaler = GradScaler()
+        if params["clip_grad"] is not None:
+            scaler = GradScaler()
+        else:
+            scaler = torch.cuda.amp.GradScaler()
```
Comment (Geeks-Sid) [inline GANDLF/training_loop.py:145]: The GradScaler handles this automatically, this is not needed anymore. Please fix this.

**Review by Geeks-Sid — CHANGES_REQUESTED** [review]

File: GANDLF/training_loop.py (hunk 2 above)
Comment (sarthakpati) [reply GANDLF/training_loop.py:145]: Actually, if that's the case, we should make a minor change for the if-loop later on, as well. Thanks, fixing it now.

Comment (sarthakpati) [reply GANDLF/training_loop.py:145]: Check in #89
